styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -171,7 +193,9 @@ private void ensureSecret(WorkflowInstance workflowInstance, RunSpec runSpec) {
                                     .build());
         LOG.info("[AUDIT] Secret {} created for service account {}", secretName, serviceAccount);
       }
-    } else if (runSpec.secret().isPresent()) {
+    }
+
+    if (runSpec.secret().isPresent()) {
```

Comment:
@fabriziodemaria @bergman Get everybody here since the decision affects internal code as well. Having to change both workflow config and container at the same time is a good argument.